api/sixtyfps-cpp/include/sixtyfps_sharedvector.h

```diff
@@ -127,6 +127,20 @@ struct SharedVector
         inner->size++;
     }
 
+    /// Clears the vector and removes all elements. The capacity remains unaffected.
+    void clear()
+    {
+        if (inner->refcount != 1) {
+            *this = SharedVector();
+        } else {
+            auto b = begin(), e = end();
+            inner->size = 0;
+            for (auto it = b; it < e; ++it) {
+                it->~T();
```

Comment:
(there is a leak when the destructor throws an exception, but let's ignore that :-)